homeassistant/components/hunterdouglas_powerview/__init__.py

```diff
@@ -58,46 +42,58 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     config = entry.data
 
     hub_address = config[CONF_HOST]
+    api_version = config.get(CONF_API_VERSION, None)
+    _LOGGER.debug("Connecting %s at %s with v%s api", DOMAIN, hub_address, api_version)
+
     websession = async_get_clientsession(hass)
 
-    pv_request = AioRequest(hub_address, loop=hass.loop, websession=websession)
+    pv_request = AioRequest(
+        hub_address, loop=hass.loop, websession=websession, api_version=api_version
+    )
 
     try:
         async with asyncio.timeout(10):
-            device_info = await async_get_device_info(pv_request, hub_address)
+            hub = Hub(pv_request)
+            await hub.query_firmware()
+            device_info = await async_get_device_info(hub)
+            if hub.role != "Primary":
+                raise IntegrationError(
+                    f"{hub.name} ({hub.hub_address}) is performing role of {hub.role} Hub. Only the Primary Hub can manage shades"
+                )
```

Comment:
We should log an error and return `False` here since the Home Assistant code will not know what to do with this exception.

Ideally we can remove this and never get here because the config flow should refuse to create the entry if its not the primary hub.

Reply:
that makes sense - should i remove this completely and move the check to the ``validate_input`` in the ``config_flow.py`` now?

Reply:
exactly 👍